skyrim/core/skyrim.py

```diff
@@ -28,24 +30,53 @@ def __init__(
             raise ValueError(f"Invalid model name(s): {missing_names}")
 
         self.model_names = model_names
+        self.ic_source = ic_source
+        self.model: GlobalEnsemble | GlobalModel
 
         if len(model_names) > 1:
             logger.info(f"Initializing ensemble model with {model_names}")
             self.model = GlobalEnsemble(model_names)
 
         else:
             self.model = MODEL_FACTORY[model_names[0]][0](ic_source=ic_source)
-        logger.debug(
-            f"Initialized {self.model} model with initial conditions from {ic_source}"
-        )
+        logger.debug(f"Initialized {self.model} model with initial conditions from {ic_source}")
 
     def __repr__(self) -> str:
-        return f"Skyrim({self.model_name})"
+        return f"Skyrim(models={self.model_names},ic={self.ic_source})"
 
     @staticmethod
     def list_available_models():
         return list(MODEL_FACTORY.keys())
 
+    def forecast(
+        self,
+        start_time: datetime.datetime,
+        lead_time: int = 6,
+        save: bool = False,
+        save_config: dict = {},
+    ) -> xr.DataArray | xr.Dataset | list[xr.DataArray | xr.Dataset] | list[str]:
```

Comment:
lets make this one return only `xr.DataArray`? Otherwise a bit too overloaded with different return types. What is the use of this function?

Reply:
this is because of graphcast, watch my next PR.